### PR TITLE
docs: Improve shell tips.

### PR DESCRIPTION
**Pull request by trueskawka**
Added some suggestions for improving the shell tips doc.

I've rearranged some sections, as it seemed more in line with the general flow of the document, but might not be necessary.

Reviews welcome.

### CODE REVIEW

File: docs/shell-tips.md [hunk 1]
```diff
@@ -1,83 +1,64 @@
...
 If you haven't used it before, you should probably take a look at
-[this tutorial](http://linuxcommand.org/lc3_learning_the_shell.php). If you're
+[this tutorial](http://linuxcommand.org/lc3_learning_the_shell.php).
+
+If you're
 using Windows,
 [these videos](https://www.youtube.com/playlist?list=PL6gx4Cwl9DGDV6SnbINlVUd0o2xT4JbMu)
 may be useful too, but keep in mind that the following tips only apply to
-Linux/macOS environments (Unix-like shells).
+Linux/macOS environments (Unix shells). You can also use a Unix-like shell
```
Comment (YagoGG) [inline docs/shell-tips.md]: Shouldn't this be "Unix shell"? ;P

File: docs/shell-tips.md [hunk 2]
```diff
@@ -1,83 +1,64 @@
...
+*terminal* (sometimes called a *console*). This is how most terminal windows
+look like:
 
 ![An example shell window](shell-screenshot.png)
 
 If you haven't used it before, you should probably take a look at
-[this tutorial](http://linuxcommand.org/lc3_learning_the_shell.php). If you're
+[this tutorial](http://linuxcommand.org/lc3_learning_the_shell.php).
+
+If you're
```
Comment (YagoGG) [inline docs/shell-tips.md]: Is this newline intended?

File: docs/shell-tips.md (hunk 1 above)
Comment (tommyip) [inline docs/shell-tips.md]: How come the -like part is removed? Linux is technical not under Unix, at least it is not Unix certified.

Comment (YagoGG) [inline docs/shell-tips.md]: But, apparently, the OS is Unix-like, not the shell. I didn't know it, either. :man_shrugging:

File: docs/shell-tips.md [hunk 3]
```diff
@@ -1,83 +1,64 @@
...
+with a dollar sign `$` or a dollar sign preceded by some text
 (e.g. `(zulip-venv)vagrant@vagrant-base-trusty-amd64:~$`).
 
 This is called the **prompt**, and it's only an indicator that the shell is
-awaiting new orders. You shouldn't type that, since it isn't part of the
-commands.
-
-## Running commands as root (`sudo`)
+awaiting new orders. The prompt can contain useful information, let's look
+at `(zulip-venv)vagrant@vagrant-base-trusty-amd64:~$`:
```
Comment (YagoGG) [inline docs/shell-tips.md]: I don't know if leaving the Vagrant prompt could lead to confusion if we're going to break it down.

Maybe we could use a different one, e.g. `(zulip-venv)john@laptop:~$`, so when the students see Vagrant's, they deduce the diffrenent fields. What do you think?

File: docs/shell-tips.md (hunk 1 above)
Comment (trueskawka) [inline docs/shell-tips.md]: I'm not sure Cygwin is Unix.

File: docs/shell-tips.md (hunk 2 above)
Comment (trueskawka) [inline docs/shell-tips.md]: yes

File: docs/shell-tips.md (hunk 3 above)
Comment (trueskawka) [inline docs/shell-tips.md]: Agreed, though I don't think you'd have `(zulip-venv)` on your machine, it's more likely to just be a `venv`.

File: docs/shell-tips.md (hunk 2 above)
Comment (YagoGG) [inline docs/shell-tips.md]: I mean the one between lines 12 and 13. Why breaking there?

File: docs/shell-tips.md (hunk 3 above)
Comment (YagoGG) [inline docs/shell-tips.md]: Yup, that's even better :D

File: docs/shell-tips.md (hunk 2 above)
Comment (trueskawka) [inline docs/shell-tips.md]: oh, sorry, this one actually shouldn't be there, I thought it was before a link

File: docs/shell-tips.md (hunk 1 above)
Comment (trueskawka) [inline docs/shell-tips.md]: Ok, so `Bash`, which is the shell used by default in most Linux and OS X distributions, is a Unix shell (https://www.wikiwand.com/en/Unix_shell), whereas Cygwin, which is a Windows tool, is a Unix-like environment with a Unix-like shell (https://www.wikiwand.com/en/Cygwin). Does that clear it up @tommyip? Already discussed it on Zulip with @YagoGG.

**Pull request merged**